app/http.php

```diff
@@ -0,0 +1,366 @@
+
+    $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    foreach ($executors as $executor) {
+        $state = $executorStates->exists($executor) ? $executorStates->get($executor) : null;
+
+        if ($state === null) {
+            Console::warning('Executor ' . $executor . ' has unknown state.');
+        } else {
+            Console::log('Executor ' . $executor . ' is ' . ($state['status'] ?? 'unknown') . '.');
+        }
+    }
+});
+
+Swoole\Event::wait();
+
+// Wildcard action
+App::error()
+    ->inject('request')
+    ->inject('response')
+    ->inject('roundRobinAtomic')
+    ->inject('executorStates')
+    ->action(function (Request $request, Response $response, Atomic $roundRobinAtomic, Table $executorStates) {
+        $secretKey = \explode(' ', $request->getHeader('authorization', ''))[1] ?? '';
+
+        if (empty($secretKey)) {
+            throw new Exception('Incorrect proxy key.', 401);
+        }
+        if ($secretKey !== App::getEnv('OPEN_RUNTIMES_PROXY_SECRET', '')) {
+            throw new Exception('Incorrect proxy key.', 401);
+        }
+
+        $roundRobinIndex = $roundRobinAtomic->get();
+
+        $algoType = App::getEnv('OPEN_RUNTIMES_PROXY_ALGORITHM', '');
+        $algo = match ($algoType) {
+            'round-robin' => new RoundRobin($roundRobinIndex - 1), // Atomic indexes from 1. Balancing library indexes from 0. That's why -1
+            'random' => new Random(),
+            default => new Random()
+        };
+
+        $balancing = new Balancing($algo);
+
+        $balancing->addFilter(fn ($option) => $option->getState('status', 'offline') === 'online');
+
+        foreach ($executorStates as $state) {
+            $balancing->addOption(new Option($state));
+        }
+
+        $body = \json_decode($request->getRawPayload() ? $request->getRawPayload() : '{}', true);
+        $runtimeId = $body['runtimeId'] ?? null;
+        // TODO: @Meldiron Use RuntimeID in CPU-based adapter
+
+        $option = $balancing->run();
+
+        if ($option === null) {
+            throw new Exception('No online executor found', 404);
+        }
+
+        if ($algo instanceof RoundRobin) {
+            $roundRobinAtomic->cmpset($roundRobinIndex, $algo->getIndex() + 1); // +1 because of -1 earlier above
+        }
+
+        $executor = [];
+
+        foreach ($option->getStateKeys() as $key) {
+            $executor[$key] = $option->getState($key);
+        }
+
+        Console::success('Executing on ' . $executor['hostname']);
+
+        $client = new Client($executor['hostname'], 80);
+        $client->setMethod($request->getMethod());
+
+        $headers = \array_merge($request->getHeaders(), [
+            'authorization' => 'Bearer ' . App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTOR_SECRET', '')
+        ]);
+
+        // Header used for testing
+        $isProduction = App::getEnv('OPEN_RUNTIMES_PROXY_ENV', 'development') === 'production';
```

Comment:
We can set the production mode and use utopia getters at the very beginning: 

Resource 1: 
https://github.com/utopia-php/framework/blob/master/src/App.php#L366-L389

Resource 2:
https://github.com/utopia-php/framework/blob/master/src/App.php#L274-L291

Resource 3:
https://github.com/appwrite/appwrite/blob/master/app/init.php#L175

Reply:
Applied.

Reply:
Also to logger, we did similar thing there.